File: Data Structures & Algorithms/counting-bits/submission-0.java
class Solution {
    public int[] countBits(int n) {
        int[] res = new int[n + 1];
        for (int i = 0; i < res.length; i++) {
            int bits = 0;
            if (i == 0) {
                res[i] = 0;
            } else {
                int temp = i;
                while (temp != 0) {
                    temp = temp & (temp - 1);
                    bits++;
                }
                res[i] = bits;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-0.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        
        res = []
        first, second = 0, 0
        while first < len(nums1) and second < len(nums2):
            if nums1[first] < nums2[second]:
                res.append(nums1[first])
                first += 1
            else:
                res.append(nums2[second])
                second += 1

        if first < len(nums1):
            for i in range(first, len(nums1)):
                res.append(nums1[i])
        
        if second < len(nums2):
            for i in range(second, len(nums2)):
                res.append(nums2[i])

        # now we have a sorted res array
        # find median in this new array
        index = (len(res) - 1) // 2
        if len(res) % 2: # is odd
            return res[index]
        else:
            return (res[index] + res[index + 1]) / 2